tests/routes/users.test.js

```diff
@@ -0,0 +1,184 @@
+const app = require('../../src/index'); // link to the server file
+const supertest = require('supertest');
+const request = supertest(app);
+
+const mongoose = require('mongoose');
+const User = require('../../src/models/user');
+
+const config = require('../../src/configs/config');
+
+const { db: { host, port, name } } = config;
+const connectionString = `mongodb://${host}:${port}/${name}`;
+
+beforeAll(async () => {
+    // disconnect mongoose from the db of app instance
+    await mongoose.disconnect();
+    await mongoose.connect(connectionString, {
+        useNewUrlParser: true
+    });
+
+    const presetUsers = [{
+            _id: '6088d0ccf5a4bb73ed24cf1f',
+            name: 'Julia Shaam',
+            dob: '1985-09-01',
+            address: '349 Cyder Street',
+            description: 'My dad is British.'
+        },
+        {
+            _id: '6088d0ccf5a4bb73ed24cf2f',
+            name: 'Brandy King',
+            dob: '1988-10-20',
+            address: '19 Rother Street',
+            description: 'I have ginger hair.'
+        },
+        {
+            _id: '6088d0ccf5a4bb73ed24cf3f',
+            name: 'Marcelo Saciloto',
+            dob: '1988-07-01',
+            address: '73 Maple Street',
+            description: 'I am a Brazilian papa.'
+        }
+    ]
+    for (const u of presetUsers) {
+        const user = new User(u);
+        await user.save();
+    }
+})
+
+
+afterAll(async () => {
+    // clean up DB.
+    await User.deleteMany();
```

Comment:
some people clean up DB before tests run, can you compare these two approaches?
also some people use `beforeEach` not `beforeAll` to do DB cleanup, can you compare these two approaches?

Reply:
I think you're right, cleaning up DB before tests run makes more sense because it could potentially clean up test data from previous tests left in the DB. 

Well, if using `beforeAll`, you need to clean up DB in the individual test, if there is some operations on DB in the test. Whereas using `beforeEach` can be less efficient than `beforeAll` because it cleans up and sets up DB more frequently, but it's also less error-prone.